### DISCUSSION TITLE
Project Files gone

### DISCUSSION THREAD

**Discussion by chrysenteron** [opengisch/QField#3838] [Bugs]
When I opened my project in the internet browser today, none of my project files showed up in "Files". Is there a general problem with the service or is it just my particular project? And will you be able to restore the files? 
Thanks in advance.

**Comment by Forstrevier** [comment]
Same thing with me. 2 projects, in both all files have disappeared since yesterday.

**Comment by suricactus** [comment]
@chrysenteron and @Forstrevier we took your input so seriosly we took down the service until we find out what is going wrong. Please share all details what you have done before you find out the project files are missing.

**Reply by chrysenteron** [reply]
Most recently I have opened the project several times either in QGIS or the internet browser to look up some data, but without synchronizing or changing any of the data. Last update to the cloud was on the 18th where I manually updated the data of a vector layer in QGIS and then synchronized it with the cloud.
My username is elias_decker and the project is called "Tabletprojekt" by the organization "Boden-mit-Zukunft"

**Comment by suricactus** [comment]
Please share your username and project name for me to be helpful.

**Comment by tildigg** [comment]
When will the site be back up? Thanks

**Reply by mbernasocchi** [reply]
we're investigating the issue and want to take all possible precaution to avoid the issue described here

**Reply by mbernasocchi** [reply]
https://status.qfield.org/incidents/60bd1643-7ade-4572-8e54-d0f7a4625cc8

**Reply by mbernasocchi** [reply]
A fix has been deployed yesterday; all systems are operational again and we are monitoring the situation.

**Comment by mbernasocchi** [comment]
Merging with https://github.com/opengisch/QField/discussions/3829

**Comment by dhmtb** [comment]
I experienced the same problem. All files are gone from my project.

**Comment by mbernasocchi** [comment]
Dear QFieldCloud BETA user, 
Thank you for reporting this invaluable information for improving QFieldCloud BETA.

As mentioned above, as a result of your report, we found an issue that could lead to data deletion under some specific circumstances. We have already identified the problem and have deployed a fix.
The issue required multiple preconditions to trigger and affected only file storage (fieldwork deltas are stored in a DB), so the data loss is minimal.

## What did we do

As soon as the first report arrived, we paused QFieldCloud BETA to protect all users and started investigating the problem and working on a fix.
After extensive testing, we deployed the fix and restarted QFieldCloud.

## What can you do

Since you may have been affected by this issue, the easiest way to get your project back online is to simply re-synchronise your local project version to QFieldCloud. Only the file storage was affected, so once you re-push your files, all will work as before. 

You can also use the [send-to functionality](https://docs.qfield.org/how-to/projects/#send-to) to get the latest data from your Android device.

The file storage backup system of QFieldCloud was being finalised for production, and part of the affected data is available there. Additionally, all the fieldwork data you collected via QField is stored in a database and is thus not affected by the issue.

If you cannot restore the missing files from your local data, please contact us at beta-transition@qfield.cloud, and we will be happy to help you out.

##  What are we doing better

According to the original plan for the upcoming general availability of QFieldCloud, we continue ramping up the file storage backup system to the production state of daily backups and cross-region replication. The database system, where all the project configuration and fieldwork data are stored, has already been regularly backed up since the beginning of public BETA. 

Along with fixing the issue, we also did an in-depth review of all deletion-related code. The fix we deployed now brings additional safeguards and more tests.

We’re sorry for the inconvenience.
Marco
